File: api-tester/src/main/java/com/revenuecat/apitester/java/PurchasesErrorAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.PurchasesError;
import com.revenuecat.purchases.PurchasesErrorCode;

@SuppressWarnings({"unused"})
final class PurchasesErrorAPI {
    static void check(final PurchasesError error) {
        final PurchasesErrorCode code = error.getCode();
        final String message = error.getMessage();
        final String underlyingErrorMessage = error.getUnderlyingErrorMessage();
    }

    static void check(final PurchasesErrorCode code) {
        switch (code) {
            case UnknownError:
            case PurchaseCancelledError:
            case StoreProblemError:
            case PurchaseNotAllowedError:
            case PurchaseInvalidError:
            case ProductNotAvailableForPurchaseError:
            case ProductAlreadyPurchasedError:
            case ReceiptAlreadyInUseError:
            case InvalidReceiptError:
            case MissingReceiptFileError:
            case NetworkError:
            case InvalidCredentialsError:
            case UnexpectedBackendResponseError:
            case InvalidAppUserIdError:
            case OperationAlreadyInProgressError:
            case UnknownBackendError:
            case InvalidAppleSubscriptionKeyError:
            case IneligibleError:
            case InsufficientPermissionsError:
            case PaymentPendingError:
            case InvalidSubscriberAttributesError:
            case LogOutWithAnonymousUserError:
            case ConfigurationError:
            case UnsupportedError:
            case EmptySubscriberAttributesError:
            case CustomerInfoError:
            case SignatureVerificationError:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PurchasesErrorAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.PurchasesError
import com.revenuecat.purchases.PurchasesErrorCode

@Suppress("unused", "UNUSED_VARIABLE")
private class PurchasesErrorAPI {
    fun check(error: PurchasesError) {
        val code: PurchasesErrorCode = error.code
        val message: String = error.message
        val underlyingErrorMessage: String? = error.underlyingErrorMessage
    }

    fun check(code: PurchasesErrorCode) {
        when (code) {
            PurchasesErrorCode.UnknownError,
            PurchasesErrorCode.PurchaseCancelledError,
            PurchasesErrorCode.StoreProblemError,
            PurchasesErrorCode.PurchaseNotAllowedError,
            PurchasesErrorCode.PurchaseInvalidError,
            PurchasesErrorCode.ProductNotAvailableForPurchaseError,
            PurchasesErrorCode.ProductAlreadyPurchasedError,
            PurchasesErrorCode.ReceiptAlreadyInUseError,
            PurchasesErrorCode.InvalidReceiptError,
            PurchasesErrorCode.MissingReceiptFileError,
            PurchasesErrorCode.NetworkError,
            PurchasesErrorCode.InvalidCredentialsError,
            PurchasesErrorCode.UnexpectedBackendResponseError,
            PurchasesErrorCode.InvalidAppUserIdError,
            PurchasesErrorCode.OperationAlreadyInProgressError,
            PurchasesErrorCode.UnknownBackendError,
            PurchasesErrorCode.InvalidAppleSubscriptionKeyError,
            PurchasesErrorCode.IneligibleError,
            PurchasesErrorCode.InsufficientPermissionsError,
            PurchasesErrorCode.PaymentPendingError,
            PurchasesErrorCode.InvalidSubscriberAttributesError,
            PurchasesErrorCode.LogOutWithAnonymousUserError,
            PurchasesErrorCode.ConfigurationError,
            PurchasesErrorCode.UnsupportedError,
            PurchasesErrorCode.EmptySubscriberAttributesError,
            PurchasesErrorCode.CustomerInfoError,
            PurchasesErrorCode.SignatureVerificationError
            -> {}
        }.exhaustive
    }
}
